packages/gatsby/src/utils/webpack.config.js

```diff
@@ -897,6 +898,19 @@ module.exports = async (
     config.cache = cacheConfig
   }
 
+  if (
+    stage === `develop` &&
+    process.env.GATSBY_EXPERIMENTAL_WEBPACK_LAZY_COMPILATION
+  ) {
+    config.experiments = {
+      lazyCompilation: {
+        entries: false,
+        imports: true,
+      },
+    }
+    config.cache.compression = `brotli`
```

Comment:
FYI in my measurements cache compression only impacted the size of the filesystem representation but not the in-memory representation.